tensorflow/compiler/xla/service/all_reduce_reassociate.cc

```diff
@@ -57,16 +108,31 @@ StatusOr<bool> AllReduceReassociate::Run(
   bool changed = false;
   for (auto computation : module->computations(execution_threads)) {
     for (HloInstruction *inst : computation->MakeInstructionPostOrder()) {
+      HloInstruction *convert0 = nullptr;
+      HloInstruction *convert1 = nullptr;
```

Comment:
Overall, I think this transformation increases the numerical accuracy, and should be controlled by a flag similar to what we have for say `xla_gpu_simplify_all_fp_conversions`. May be a flags like `xla_gpu_allow_ar_promotion_for_reassociation` or something little less verbose if possible. If the flag is turned off, we should not match the converts.

Reply:
agree that it'd be better, added an option to control this and default it to true